syntax = "proto3";

message Post {
    int32 id = 1;
    string title = 2;
    string content = 3;
    string timestamp = 4;
    int32 user_id = 5;
}

message CreatePostRequest {
    int32 id = 1;
    string title = 2;
    string content = 3;
    string timestamp = 4;
    int32 user_id = 5;
}

message CreatePostResponse {
    Post post = 1;
}

message ReadPostRequest {
    int32 id = 1;
}

message ReadPostResponse {
    Post post = 1;
}

service PostService {
    rpc CreatePost(CreatePostRequest) returns (CreatePostResponse);
    rpc ReadPost(ReadPostRequest) returns (ReadPostResponse);
}
